### DISCUSSION TITLE
Align with java-diff-utils 4.12

### DISCUSSION THREAD

**Discussion by lppedd** [petertrr/kotlin-multiplatform-diff#101] [General]
Hey! Thanks for this KMP library.

I'm planning to adopt it to batch saving editor changes. That is, at save time - after diffing the two states - I'm going to collect all inserts, all deletions, and all edits, and perform the necessary operations to persist those changes. Sounds crazy but there is always a reason.

To do this I'd like to use the Myers algorithm in linear space, which this library is currently lacking.  
I think the best path would be to have you adopt the upstream changes, since you know where you left the project.

I'll then experiment with adopting it, I'll evaluate performance, and come back with PRs if necessary, like [what I did for antlr-kotlin](https://github.com/Strumenta/antlr-kotlin/graphs/contributors) (although in that case I did also port the upstream changes).

Or would you prefer I do everything? Let me know if you don't have the time.

**Comment by petertrr** [comment]
Hey! TBH I don't have that much time at the moment. I'll be happy to review PRs if you'd like to submit any, though.

**Comment by lppedd** [comment]
We can close as it's been done.
